py/frequenz/client/weather/_client.py

```diff
@@ -0,0 +1,55 @@
+import grpc
+from frequenz.api.weather import weather_pb2_grpc, weather_pb2
+from ._types import Forecasts, ForecastFeature, Location
+from frequenz.channels import Receiver
+from frequenz.client.base.grpc_streaming_helper import GrpcStreamingHelper
+
+
+class Client:
+    """Weather forecast client."""
+
+    def __init__(self, grpc_channel: grpc.aio.Channel, svc_addr: str) -> None:
+        """Initialize the client.
+
+        Args:
+            grpc_channel: gRPC channel to use for communication with the API.
+            svc_addr: Address of the service to connect to.
+        """
+        self._svc_addr = svc_addr
+        self._stub = weather_pb2_grpc.WeatherForecastServiceStub(grpc_channel)
+        self._streams: dict[
+            tuple[list[Location], list[ForecastFeature]],
+            GrpcStreamingHelper[
+                weather_pb2.ReceiveLiveWeatherForecastResponse, Forecasts
+            ],
+        ] = {}
+
+    async def stream_live_forecast(
+        self,
+        locations: list[Location],
+        features: list[ForecastFeature],
+    ) -> Receiver[Forecasts]:
+        """Stream live weather forecast data.
+
+        Args:
+            locations: locations to stream data for.
+            features: features to stream data for.
+            maxsize: maximum number of messages to buffer.
+
+        Returns:
+            Async generator of weather forecast data.
+        """
+
+        stream_key = (locations, features)
```

Comment:
Lists cannot be used as key, but you can do `tuple(locations + features)` instead.

Reply:
true, I actually don't like that the client creates new streams just because of a different order. What do you think about sorting the lists and create the key afterwards as mentioned above? The order would then actually depend on the first request but can be easily changed through the filtering.
what do you think? @cwasicki @shsms

Reply:
We don't rely on the ordering, right? Then this could be a good option.

Reply:
Although, I think this could lead to errors if users expect they know the order. I am in favor of keeping it like it is now and just fix the tuple bug.

Reply:
looks like this is done.